src/formatDate.js

```diff
@@ -50,7 +50,31 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedDate = date.split(fromFormat[3]);
+  const dataObj = {};
+  let resultString = '';
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    dataObj[fromFormat[i]] = splitedDate[i];
+  }
+
+  if (!dataObj.YY) {
+    dataObj.YY = dataObj.YYYY.split('').slice(2, 5).join('');
```

Comment:
String has [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) method too
Don’t need convert it to array and after to string